### PR TITLE
added unlink, unlinkat syscalls

### PR DESCRIPTION
**Pull request by rksharma95**
added sycalls `unlink` and `unlinkat` to the bpf module to get logs.

### CODE REVIEW

**Comment by nyrahul** [comment]
Thanks @rksharma95 .. Can you please attach the same telemetry events generated with unlink/unlinkat?

**Comment by rksharma95** [comment]
here are the telemetry events generated for a sequence of commands executed inside the pod monitored by kubearmor:

```
// inside pod
$ touch testFile.txt
$ rm testFile.txt
$ mkdir testDir
$ rm -r testDir 
```
telemetry events (unlink/unlinkat syscall events only)
```json
{"Timestamp":1655205972,
"UpdatedTime":"2022-06-14T11:26:12.414264Z",
"ClusterName":"default",
"HostName":"kubearmor-dev",
"NamespaceName":"multiubuntu",
"PodName":"ubuntu-1-deployment-5bd8d67678-8tgph",
"Labels":"container=ubuntu-1,group=group-1",
"ContainerID":"159c717ff7bc9f9df51ea01b9250c0e3492001ea26d2632e119f5115a3233729",
"ContainerName":"ubuntu-1-container",
"ContainerImage":"kubearmor/ubuntu-w-utils:0.1@sha256:b4693b003ed1fbf7f5ef2c8b9b3f96fd853c30e1b39549cf98bd772fbd99e260",
"ParentProcessName":"/bin/bash",
"ProcessName":"/bin/rm",
"HostPPID":8433,
"HostPID":8853,
"PPID":251,
"PID":261,
"Type":"ContainerLog",
"Source":"/bin/rm testFile.txt",
"Operation":"File",
"Resource":"testFile.txt",
"Data":"syscall=SYS_UNLINKAT fd=-100 flags=","Result":"Passed"}

{"Timestamp":1655206012,
"UpdatedTime":"2022-06-14T11:26:52.135853Z",
"ClusterName":"default",
"HostName":"kubearmor-dev",
"NamespaceName":"multiubuntu",
"PodName":"ubuntu-1-deployment-5bd8d67678-8tgph",
"Labels":"container=ubuntu-1,group=group-1",
"ContainerID":"159c717ff7bc9f9df51ea01b9250c0e3492001ea26d2632e119f5115a3233729",
"ContainerName":"ubuntu-1-container",
"ContainerImage":"kubearmor/ubuntu-w-utils:0.1@sha256:b4693b003ed1fbf7f5ef2c8b9b3f96fd853c30e1b39549cf98bd772fbd99e260",
"ParentProcessName":"/bin/bash",
"ProcessName":"/bin/rm",
"HostPPID":8433,
"HostPID":8917,
"PPID":251,
"PID":264,
"Type":"ContainerLog",
"Source":"/bin/rm -r testDir",
"Operation":"File",
"Resource":"testDir",
"Data":"syscall=SYS_UNLINKAT fd=-100 flags=AT_REMOVEDIR","Result":"Passed"}

```

**Comment by nyrahul** [comment]
Why is the fd value -100 in the event details?

**Comment by nyrahul** [comment]
Can you please squash the commits into a single one? Thanks

**Comment by nyrahul** [comment]
@rksharma95 , can you remove the print of fd value from the event?

**Comment by nyrahul** [comment]
> @rksharma95 , can you remove the print of fd value from the event?

Looks like it is already removed. The telemetry events are based on old code I guess!

**Comment by rksharma95** [comment]
> > @rksharma95 , can you remove the print of fd value from the event?
> 
> Looks like it is already removed. The telemetry events are based on old code I guess!

yes, the `fd` value has been removed.

**Comment by rksharma95** [comment]
> Why is the fd value -100 in the event details?

it's a special value of `fd`, fd=-100 or `AT_FDCWD` states that pathname should be interpreted relative to the current working directory of the calling process.

File: KubeArmor/BPF/system_monitor.c [hunk 1]
```diff
@@ -762,6 +768,25 @@ static __always_inline int events_perf_submit(struct pt_regs *ctx)
 
 // == Full Path == //
 
+SEC("kprobe/security_path_unlink")
+int kprobe__security_path_unlink(struct pt_regs *ctx){
+    if (skip_syscall())
+		return 0;
+
+	struct path *dir = (struct path*) PT_REGS_PARM1(ctx);
+    struct dentry *dentry = (struct dentry*) PT_REGS_PARM2(ctx);
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
    struct path *dir = (struct path*) PT_REGS_PARM1(ctx);
    struct dentry *dentry = (struct dentry*) PT_REGS_PARM2(ctx);
```

File: KubeArmor/BPF/system_monitor.c [hunk 2]
```diff
@@ -762,6 +768,25 @@ static __always_inline int events_perf_submit(struct pt_regs *ctx)
...
+    
+    struct path p;
+
+    p.dentry = READ_KERN(dentry);
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c:789]: We should have a sanity null checks for the returned pointers `dir` and `dentry`?

**Review by nyrahul — CHANGES_REQUESTED** [review]

**Comment by nyrahul** [comment]
@achrefbensaad , this needs conflict resolution. Thanks

File: KubeArmor/BPF/system_monitor.c (hunk 2 above)
Comment (achrefbensaad) [reply KubeArmor/BPF/system_monitor.c:789]: Done

**Review by nyrahul — APPROVED** [review]
LGTM. Thank you for handlling all the changes at top speed.

**Pull request merged**